subsys/bluetooth/shell/bt.c

```diff
@@ -193,65 +193,128 @@ static bool data_cb(struct bt_data *data, void *user_data)
 	}
 }
 
-static void print_data_set(uint8_t type, uint8_t set_value_len,
+static void print_data_hex(const uint8_t *data, uint8_t len, bool warning)
+{
+	shell_fprintf(ctx_shell, warning ? SHELL_WARNING : SHELL_INFO, "0x");
+	/* Reverse the byte order when printing as advertising data is LE
+	 * and the MSB should be first in the printed output.
+	 */
+	for (int8_t i = len - 1; i >= 0; i--) {
```

Comment:
If you need `i` to be signed here, then it needs to be at least a `int16_t` to hold all `uint8_t` values